src/Infrastructure.Styles.Analyzer.Tests/RMSYN1103ExtensionMethodsTestedAsNormalInvocationAnalyzerTests.cs

```diff
@@ -0,0 +1,329 @@
+﻿// This file is part of the re-motion Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// re-motion Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+
+using Gu.Roslyn.Asserts;
+using NUnit.Framework;
+
+
+[assembly:Gu.Roslyn.Asserts.TransitiveMetadataReferences(typeof(NUnitAttribute))]
+namespace Infrastructure.Styles.Analyzer.Tests
+{
+    
+    [TestFixture]
+    public class RMSYN1103ExtensionMethodsTestedAsNormalInvocationAnalyzerTests
+    {
+        private static readonly RMSYN1103ExtensionMethodsTestedAsNormalInvocationSyntaxAnalyzer Analyzer = new RMSYN1103ExtensionMethodsTestedAsNormalInvocationSyntaxAnalyzer();
+
+        [Test]
+        public void Analyze_WithExtensionMethodAndAnnotation_ReportsDiagnostic ()
+        {
+            var code = @"
+using NUnit.Framework;
+
+namespace N
+{
+  public static class D
+  {
+    public static void ibimsafunktion(this int a) 
+    {
+    }
+  }
+  
+  class C
+  {
+    [Test]
```

Comment:
I think we are missing a test with `[TestAttribute]` (the optional Attribute suffix?)

Also a test with the extension method in a block/if statement would be great